daemon/cluster/convert/service.go

```diff
@@ -272,7 +272,7 @@ func restartPolicyToGRPC(p *types.RestartPolicy) (*swarmapi.RestartPolicy, error
 	var rp *swarmapi.RestartPolicy
 	if p != nil {
 		rp = &swarmapi.RestartPolicy{}
-		sanatizedCondition := strings.ToUpper(strings.Replace(string(p.Condition), "-", "_", -1))
+		sanatizedCondition := strings.ToUpper(p.Condition.String())
```

Comment:
But from the Doc, it shows "on-failure".
I use "on-failure" to update service , but I got "on_failure".